src/lib/pickles/wrap_verifier.ml

```diff
@@ -207,14 +207,14 @@ struct
   let lagrange (type n)
       ~domain:
         ( (which_branch : n One_hot_vector.t)
-        , (domains : (Domains.t, n) Vector.t) ) i =
+        , (domains : (Domains.t, n) Vector.t) ) srs i =
     Vector.map domains ~f:(fun d ->
-        let d =
-          Precomputed.Lagrange_precomputations.index_of_domain_log2
-            (Domain.log2_size d.h)
-        in
-        match Precomputed.Lagrange_precomputations.vesta.(d).(i) with
-        | [| g |] ->
+        let d = Int.pow 2 (Domain.log2_size d.h) in
+        match
+          (Kimchi_bindings.Protocol.SRS.Fp.lagrange_commitment srs d i)
+            .unshifted
+        with
+        | [| Finite g |] ->
```

Comment:
What does `Finite` do?

Reply:
The group elements have type `'g or_infinity = Finite of 'g | Infinity`, so that we can represent it as `(x, y)` coordinates if it isn't the point at infinity.